### PR TITLE
Move yaml to mux tests to plugin

### PR DESCRIPTION
**Pull request by clebergnu**
This moves tests, and some other code that is used exclusively by the YAML to Mux plugin, to its module directories.

This version is based on top of the parallel run for plugins commit, and was tested with both parallel and serial execution.

### CODE REVIEW

File: selftests/run [hunk 1]
```diff
@@ -7,13 +7,11 @@ import gc
 import os
 import subprocess
 import sys
-import logging
 import unittest
 
 from avocado.core import data_dir
 
 
-logger = logging.getLogger(__name__)
```
Comment (lmr) [inline selftests/run]: OK

File: selftests/unit/test_mux.py [hunk 2]
```diff
@@ -369,7 +383,8 @@ def test_get_greedy_path(self):
         # path matches nothing
         self.assertEqual(self.params1.get('root', '', 999), 999)
 
-    @unittest.skipIf(not yaml_to_mux.MULTIPLEX_CAPABLE, "Not multiplex capable")
+    @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
+                         "yaml_to_mux plugin available")
```
Comment (lmr) [inline selftests/unit/test_mux.py]: I suppose you meant `Plugin yaml_to_mux not available` here.

File: selftests/run [hunk 3]
```diff
@@ -24,6 +33,11 @@ def test_suite():
     for section in ('unit', 'functional', 'doc'):
         suite.addTests(loader.discover(start_dir=os.path.join(selftests_dir, section),
                                        top_level_dir=basedir))
+    # when other plugins get their own tests, make this a generic loop
+    if plugin_available('avocado-framework-plugin-varianter-yaml-to-mux'):
+        path = os.path.join(basedir, 'optional_plugins',
+                            'varianter_yaml_to_mux', 'tests')
+        suite.addTests(loader.discover(start_dir=path, top_level_dir=path))
```
Comment (lmr) [inline selftests/run]: Great cleanup.

File: selftests/unit/test_mux.py (hunk 2 above)
Comment (clebergnu) [reply selftests/unit/test_mux.py]: True!

Comment (clebergnu) [reply selftests/unit/test_mux.py]: I'm changing them all:

```diff
diff --git a/selftests/unit/test_mux.py b/selftests/unit/test_mux.py
index 15fb99bb..39aa5686 100644
--- a/selftests/unit/test_mux.py
+++ b/selftests/unit/test_mux.py
@@ -245,7 +245,7 @@ class TestMuxTree(unittest.TestCase):
 class TestMultiplex(unittest.TestCase):
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def setUp(self):
         self.mux_tree = yaml_to_mux.create_from_yaml(['/:' + PATH_PREFIX +
                                                       'selftests/.data/mux-selftest.'
@@ -310,14 +310,14 @@ class TestAvocadoParams(unittest.TestCase):
                                                 ['/ch1/*', '/ch0/*'])
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_pickle(self):
         params = pickle.dumps(self.params1, 2)  # protocol == 2
         params = pickle.loads(params)
         self.assertEqual(self.params1, params)
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_basic(self):
         self.assertEqual(self.params1, self.params1)
         self.assertNotEqual(self.params1, self.params2)
@@ -327,7 +327,7 @@ class TestAvocadoParams(unittest.TestCase):
         self.assertEqual(15, sum([1 for _ in iteritems(self.params1)]))
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_unhashable(self):
         """ Verifies that unhashable arguments can be passed to params.get """
         self.assertEqual(self.params1.get("root", "/ch0/", ["foo"]), ["foo"])
@@ -336,7 +336,7 @@ class TestAvocadoParams(unittest.TestCase):
                                           ['bar']), 'unique1')
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_get_abs_path(self):
         # /ch0/ is not leaf thus it's not queryable
         self.assertEqual(self.params1.get('root', '/ch0/', 'bbb'), 'bbb')
@@ -359,7 +359,7 @@ class TestAvocadoParams(unittest.TestCase):
                                           'hhh'), 'hhh')
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_get_greedy_path(self):
         self.assertEqual(self.params1.get('unique1', '/*/*/*/ch0.1.1.1/',
                                           111), 'unique1')
@@ -384,7 +384,7 @@ class TestAvocadoParams(unittest.TestCase):
         self.assertEqual(self.params1.get('root', '', 999), 999)
 
     @unittest.skipUnless(plugin_available("avocado-framework-plugin-varianter-yaml-to-mux"),
-                         "yaml_to_mux plugin available")
+                         "Plugin avocado_varianter_yaml_to_mux not available")
     def test_get_rel_path(self):
         self.assertEqual(self.params1.get('root', default='iii'), 'root')
         self.assertEqual(self.params1.get('unique1', '*', 'jjj'), 'unique1')
```

**Comment by clebergnu** [comment]
Closing in favor of #2376

**Pull request closed without merging**